cobalt/browser/application.cc

```diff
@@ -1052,6 +1059,10 @@ Application::~Application() {
   // for the debugger to land.
   watchdog::Watchdog::DeleteInstance();
 
+  // Explicitly delete the global metrics services manager here to give it
+  // an opportunity to clean up late logs and persist metrics.
+  delete metrics_services_manager_;
```

Comment:
This seems out of place with overall style - raw pointer deletes always do. All other owned modules here have `module.reset()` calls to clean them up - see `browser_module_` and `network_module_` below.

If it comes via a `GetInstance()` call from a singleton-like thing above, the symmetric thing to do would be a `metrics::CobaltMetricsServicesManager::ReleaseInstance(metrics_services_manager_)` or `DestroyInstance` perhaps ?

Reply:
Added a DeleteInstance method to CobaltMetricsServicesManager.